dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/appstore/DefaultAppStoreService.java

```diff
@@ -53,34 +55,47 @@
 /**
  * Created by zubair@dhis2.org on 07.09.17.
  */
-@Service( "org.hisp.dhis.appstore2.AppStoreService" )
-public class DefaultAppStoreService implements AppStoreService
+@Service( "org.hisp.dhis.appstore.AppStoreService" )
+public class DefaultAppStoreService
+    implements
+    AppStoreService
 {
     private static final Log log = LogFactory.getLog( DefaultAppStoreService.class );
 
     private final RestTemplate restTemplate;
 
     private final AppManager appManager;
 
-    private final SystemSettingManager systemSettingManager;
+    private final DhisConfigurationProvider dhisConfigurationProvider;
 
     public DefaultAppStoreService( RestTemplate restTemplate, AppManager appManager,
-        SystemSettingManager systemSettingManager )
+        DhisConfigurationProvider dhisConfigurationProvider )
     {
         checkNotNull( restTemplate );
         checkNotNull( appManager );
-        checkNotNull( systemSettingManager );
+        checkNotNull( dhisConfigurationProvider );
         this.restTemplate = restTemplate;
         this.appManager = appManager;
-        this.systemSettingManager = systemSettingManager;
+        this.dhisConfigurationProvider = dhisConfigurationProvider;
     }
 
     @Override
     public List<WebApp> getAppStore()
     {
-        String appStoreApiUrl = (String) systemSettingManager.getSystemSetting( SettingKey.APP_STORE_API_URL );
+        String appStoreApiUrl = dhisConfigurationProvider.getProperty( ConfigurationKey.APP_STORE_API_URL );
+        String appStoreApiVersion = dhisConfigurationProvider.getProperty( ConfigurationKey.APP_STORE_API_VERSION );
+        String allAppsUrl;
 
-        WebApp[] apps = restTemplate.getForObject( appStoreApiUrl, WebApp[].class );
+        if ( "" == appStoreApiVersion )
+        {
+            allAppsUrl = appStoreApiUrl + "/apps";
+        }
+        else
+        {
+            allAppsUrl = appStoreApiUrl + "/" + appStoreApiVersion + "/apps";
+        }
```

Comment:
This matches the hardcoded behaviour of today, but when the core starts talking to the app store api using new endpoints this logic should be abstracted out.